File: travel_parent/tarvel_web/src/main/java/com/itheima/travel/controller/FavoriteController.java
package com.itheima.travel.controller;

import com.itheima.travel.service.FavoriteService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;

@Controller
@RequestMapping("/favorite")
public class FavoriteController {

    @Autowired
    private FavoriteService favoriteService;

    /**
     * 判断是否收藏
     * @param rid
     * @param uid
     */
    @RequestMapping("/isFavorite.do")
    public void isFavorite(String rid, int uid) {
        boolean favorite = favoriteService.isFavorite(rid, uid);
        System.out.println(favorite);
    }


    /**
     * 添加收藏
     * @param
     * @param
     */
    @RequestMapping("/addFavorite.do")
    public void addFavorite(String rid, int uid){

        favoriteService.add(rid, uid);

    }
}

File: travel_parent/tarvel_web/src/main/webapp/js/controller/FavoriteController.js
app.controller("FavoriteController", function ($scope, FavoriteService) {

    // //判断是否收藏
    // $scope.isFavorite.success(
    //     function (response) {
    //         $scope.entity=response;
    // });


    //点击收藏
    // $scope.addFavorite.success(
    //     function (response) {
    //         $scope.entity=response;
    // });


    // //读取列表数据绑定到表单中
    // $scope.findAll=function(){
    //     brandService.findAll().success(
    //         function(response){
    //             $scope.list=response;
    //         }
    //     );
    // }

    //根据id查询一条路线
    $scope.findById=function (rid) {
        FavoriteService.findById(rid).success(
            function (response) {
                $scope = response;
            }
        )
    }
});